pages/agent/v3/hooks.md.erb

```diff
@@ -1,40 +1,40 @@
+1. [Plugin hooks](#plugin-hooks) - these are provided by any [plugins](/docs/agent/v3/plugins) you've defined in your `pipeline.yml`.
 
-Hooks are called in the following order:
+## Available hooks
 
-1. `environment` (global) - runs before all other commands, useful for [exporting secret keys](/docs/agent/v3/securing#using-environment-hooks-for-secrets) etc.
-1. `pre-checkout` (global) - runs before checkout
-1. `checkout` (global) - overrides the default `git checkout` behavior
-1. `post-checkout` (global) - runs after checkout
-1. `post-checkout` (local) - runs after checkout
-1. `pre-command` (global) - runs before the build command
-1. `pre-command` (local) - runs before the build command
-1. `command` (global) - overrides the default script running behavior, useful for changing how the command is evaluated (such as running it inside a container)
-1. `post-command` (local) - runs after the command
-1. `post-command` (global) - runs after the command
-1. `pre-artifact` (local) - runs before artifacts are uploaded, if an artifact upload pattern was defined for the job
-1. `pre-artifact` (global) - runs before artifacts are uploaded, if an artifact upload pattern was defined for the job
-1. `post-artifact` (local) - runs after artifacts have been uploaded, if an artifact upload pattern was defined for the job
-1. `post-artifact` (global) - runs after artifacts have been uploaded, if an artifact upload pattern was defined for the job
-1. `pre-exit` (local) - runs before the job finishes, useful for performing cleanup tasks
-1. `pre-exit` (global) - runs before the job finishes, useful for performing cleanup tasks
+The following is a complete list of hooks that can be implemented, and the order in which they are called:
 
-<div class="Docs__note">
-<p class="Docs__note__heading">Windows hooks</p>
-These hooks work on unix-based systems with bash in the path. If you're running Windows agents, you'll need some Windows-specific additions to get your hooks running. See <a href="hooks#hooks-on-windows">Hooks on Windows</a> for a detailed description.</p>
-</div>
+<table>
+  <thead>
+    <tr><th>Hook</th><th>Order</th></th><th>Description</th></tr>
+  </thead>
+  <tbody>
+    <tr><td style="white-space: nowrap"><code>environment</code></td><td style="white-space: nowrap">Agent, Plugin</td><td>Runs before all other commands. Useful for <a href="/docs/agent/v3/securing#using-environment-hooks-for-secrets">exposing secret keys</a> and <a href="/docs/agent/v3/securing#whitelisting">security whitelisting</a>.</td></tr>
+    <tr><td style="white-space: nowrap"><code>pre-checkout</code></td><td style="white-space: nowrap">Agent, Plugin</td><td>Runs before checkout.</td></tr>
+    <tr><td style="white-space: nowrap"><code>checkout</code></td><td style="white-space: nowrap">Agent, Plugin</td><td>Overrides the default <code>git checkout</code> behavior. <em>Note:</em> If multiple checkout hooks are found, only the first will be run.</td></tr>
```

Comment:
Should checkout be around the other way? Re: the chats in campfire just now

Reply:
Doh! Yes, yes it should! Update :) Thanks for the catch.